packages/search/src/features/search/utils.ts

```diff
@@ -223,6 +231,282 @@ export function queryBuilder(
   }
 }
 
+export function advancedQueryBuilder(params: IAdvancedSearchParam) {
+  const must: any[] = [
+    {
+      query_string: {
+        default_field: 'type',
+        query: `(${REGISTERED_STATUS}) OR (${CERTIFIED_STATUS})`
+      }
+    }
+  ]
+
+  if (params.event) {
+    must.push({
+      match: {
+        event: params.event
+      }
+    })
+  }
+
+  if (params.eventLocationId) {
+    must.push({
+      match: {
+        eventLocationId: params.eventLocationId
+      }
+    })
+  }
+
+  if (params.childFirstNames) {
+    must.push({
+      multi_match: {
+        query: params.childFirstNames,
+        fields: ['childFirstNames'],
+        fuzziness: 'AUTO'
+      }
+    })
+  }
+
+  if (params.childLastName) {
+    must.push({
+      multi_match: {
+        query: params.childLastName,
+        fields: 'childLastName',
+        fuzziness: 'AUTO'
+      }
+    })
+  }
+
+  if (!params.dateOfEventStart && !params.dateOfEventEnd && params.childDoB) {
+    must.push({
+      match: {
+        childDoB: params.childDoB
+      }
+    })
+  }
+
+  if (params.deceasedFirstNames) {
+    must.push({
+      multi_match: {
+        query: params.deceasedFirstNames,
+        fields: 'deceasedFirstNames',
+        fuzziness: 'AUTO'
+      }
+    })
+  }
+
+  if (params.deceasedFamilyName) {
+    must.push({
+      multi_match: {
+        query: params.deceasedFamilyName,
+        fields: 'deceasedFamilyName',
+        fuzziness: 'AUTO'
+      }
+    })
+  }
+
+  if (!params.dateOfEventStart && !params.dateOfEventEnd && params.deathDate) {
+    must.push({
+      match: {
+        deathDate: params.deathDate
+      }
+    })
+  }
+
+  if (params.dateOfEventStart || params.dateOfEventEnd) {
+    if (!params.event) {
+      throw new Error('Event is required for date range search')
+    }
+    if (!params.dateOfEventStart) {
+      throw new Error(
+        'dateOfEventStart must be provided along with dateOfEventEnd'
+      )
+    }
+    if (!params.dateOfEventEnd) {
+      throw new Error(
+        'dateOfEventEnd must be provided along with dateOfEventStart'
+      )
+    }
+
+    let fieldName
+    const eventWiseFieldName = {
+      Birth: 'childDoB',
+      Death: 'deathDate'
+    }
+    if (params.event in eventWiseFieldName) {
+      fieldName = eventWiseFieldName[params.event]
+    } else {
+      throw Error('Invalid Event')
```

Comment:
Let's make the error message a bit verbose
```suggestion
      throw Error('Invalid Event. Valid events are: Birth, Death')
```